TLM/TLM/TrafficLight/Impl/TimedTrafficLights.cs

```diff
@@ -1091,7 +1091,7 @@ public class TimedTrafficLights : ITimedTrafficLights {
             if (!Constants.ServiceFactory.NetService.IsSegmentValid(segmentId)) {
                 Log.ErrorIf(
                     logTrafficLights,
-                    () => $"TimedTrafficLights.ChangeLightMode: Segment {segmentId} is invalid");
+                    $"TimedTrafficLights.ChangeLightMode: Segment {segmentId} is invalid");
```

Comment:
[low priority note]: 
String.format still creates garbage. I think the segmentId is passed to params [] and allocates array. 

I think IDE should give you a warning about this anyway.

In any case this does not look like performance critical part of the code to me so its fine.

Reply:
Reverted to use format lambda, it was better this way